src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ToolStripContainerDesigner.cs

```diff
@@ -0,0 +1,502 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.ComponentModel;
+using System.Collections;
+using System.ComponentModel.Design;
+using System.Drawing;
+using System.Drawing.Design;
+using System.Windows.Forms.Design.Behavior;
+
+namespace System.Windows.Forms.Design;
+
+internal class ToolStripContainerDesigner : ParentControlDesigner
+{
+    private ToolStripPanel? _topToolStripPanel;
+    private ToolStripPanel? _bottomToolStripPanel;
+    private ToolStripPanel? _leftToolStripPanel;
+    private ToolStripPanel? _rightToolStripPanel;
+    private ToolStripContentPanel? _contentToolStripPanel;
+
+    private const string TopToolStripPanelName = "TopToolStripPanel";
+    private const string BottomToolStripPanelName = "BottomToolStripPanel";
+    private const string LeftToolStripPanelName = "LeftToolStripPanel";
+    private const string RightToolStripPanelName = "RightToolStripPanel";
+    private const string ContentToolStripPanelName = "ContentPanel";
+
+    private Control[]? _panels;
+    private bool _disableDrawGrid;
+    private ISelectionService? _selectionService;
+    private ToolStripContainer? _toolStripContainer;
+
+    /// <summary>
+    ///  The <see cref="IDesignerHost"/> that owns this designer.
+    /// </summary>
+    private IDesignerHost _designerHost => (IDesignerHost)GetService(typeof(IDesignerHost));
+
+    /// <summary>
+    ///  Shadow the <see cref="ToolStripContainer.TopToolStripPanelVisible"/> property at design-time
+    ///  so that we only set the visibility at design time if the user sets it directly.
+    /// </summary>
+    private bool TopToolStripPanelVisible
+    {
+        get
+        {
+            return (bool)ShadowProperties[nameof(TopToolStripPanelVisible)];
+        }
```

Comment:
```suggestion
        get => (bool)ShadowProperties[nameof(TopToolStripPanelVisible)];
```

The same can be applied for some other get methods in this file